### PR TITLE
Make profiles configurable for different operating systems.

### PR DESCRIPTION
**Pull request by msrb**
mtps-prepare-system executes steps that are tailored for a specific operating system (RHEL right now). It doesn't make sense to execute the same steps, or use the same repo files for other systems (like Fedora, CentOS Stream, ...)

This commit moves OS-specific bits into the "profiles/<os>/" directory.

Support for more operating systems can be added by adding a corresponding prepare script, and repo files, into the "profiles/<os>/" directory.

Signed-off-by: Michal Srb <michal@redhat.com>

### CODE REVIEW

File: profiles/rhel/prepare-system [hunk 1]
```diff
@@ -0,0 +1,62 @@
+INT_CERTS="${INT_CERTS:-http://download.eng.bos.redhat.com/brewroot/vol/rhel-7/packages/redhat-internal-cert-install/0.1/9.el7.csb/noarch/redhat-internal-cert-install-0.1-9.el7.csb.noarch.rpm}"
+
+if ! rpm -q redhat-internal-cert-install >/dev/null 2>&1; then
+    echo "Deploy internal certs."
+    yum install -y "$INT_CERTS"
```
Comment (jimbair) [inline profiles/rhel/prepare-system:5]: Should we validate against $? from the yum install? I assume the internal cert install failing would be a failure condition.

File: profiles/rhel/prepare-system [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+if [ -n "$ENABLE_BUILD_ROOT" ]; then
+    buildroot_repo=$(cat "$REPOFILE" | sed -n -e '/^\[/{/buildroot/p}' | tr -d '[]')
+    yum config-manager --set-enabled "$buildroot_repo"
+fi
+
+# repoquery util acts not on live system, but rather on set of enabled repos. mini-tps hold personal list of protected packages, such as openssh-server
+yum config-manager --set-enabled astepano-mini-tps || :
+
+echo "Installing required packages for testing"
```
Comment (jimbair) [inline profiles/rhel/prepare-system:57]: Should we validate these yum install steps as well? the install/update/clean/etc? A basic run function to validate and cleanup if a failure on an expected successful shell command could make this simple and safe (unless we're using set to abort on any failure; though it makes for messy logs generally).

File: profiles/rhel/prepare-system (hunk 1 above)
Comment (msrb) [reply profiles/rhel/prepare-system:5]: I am actually just moving this code from `mtps-prepare-script` to a separate file. So this code is not new.

But to answer your question: `prepare-system` is injected into `mtps-prepare-system`: https://github.com/fedora-ci/mini-tps/blob/8c1a2e78c61c1128b24da10fdd9af9f46755a181/mtps-prepare-system#L153
And that script has `-e` (fail on any non-zero return value) in shebang: `#!/bin/bash -efu`

Comment (jimbair) [reply profiles/rhel/prepare-system:5]: Okay, that makes sense. Thanks!

**Comment by Andrei-Stepanov** [comment]
@msrb looks good.
Please merge when you feel necessary.

**Comment by msrb** [comment]
Thanks! I will deal with those shellcheck warnings in another pull requests.

**Pull request merged**